linera-explorer/src/lib.rs

```diff
@@ -438,41 +495,57 @@ async fn route_aux(
         (Some(p), _) => (p, args.to_vec()),
         (_, p) => page_name_and_args(p),
     };
-    let address = node_service_address(&data.config, Protocol::Http);
+    let address = url(&data.config, Protocol::Http, AddressKind::Node);
     let result = match chain_info {
         Err(e) => Err(e),
         Ok((chain_id, chain_changed)) => {
             let page_result = match page_name {
```

Comment:
Would it make sense to extract this `match` into a separate function, so that you can use `?` and `bail!` inside it?